etl/steps/data/garden/democracy/2024-03-07/bmr.py

```diff
@@ -0,0 +1,770 @@
+To estimate regional-level countries, some workarounds are implemented. Note that not all workarounds apply to all tables.
+
+    - We "impute" classifications for countries that did not exist based on their historical equivalents. This serves two purposes:
+
+        1. To be able to colour these world regions in grapher map charts. (in table 'bmr')
+        2. To be able to count the number of people living in democracy (or non-democracy). (in table 'population_regime' and 'population_regime_years')
+
+        (applies to tables 'bmr', 'population_regime', 'population_regime_years')
+
+    - We ignore imputed values to avoid counting countries that did not actually exist back then. Note that when estimating the number of democracies, we only want to consider the countries that existed back then.
+
+
+ADDITIONAL FILES
+----------------
+Some additional files are provided to help out in the processing:
+
+    - 'bmr.countries_impute.yml': Specify how to impute countries (i.e. inherit values for country A from country B during certain year period Y1-Y2).
+
+    - 'bmr.countries_overlap.json': Specify which countries overlap with others when counting countries. USSR-ones are allowed since regions dataset is not accurate for them. Other states (german ones) are accepted, based on this dataset's cirterium.
+"""
+
+import json
+from typing import Any, Dict, List, Set, Tuple, Union, cast
+
+import numpy as np
+import pandas as pd
+import yaml
+from owid.catalog import Dataset, Variable
+from owid.catalog.tables import Table, concat
+
+from etl.data_helpers import geo
+from etl.helpers import PathFinder, create_dataset
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+
+
+# IMPUTE: We will infer indicator values for some countries based on their historical equivalences.
+path = paths.directory / "bmr.countries_impute.yml"
+COUNTRIES_IMPUTE = yaml.safe_load(path.read_text())
+
+# Overlapping countries as expected when counting countries
+COUNTRIES_OVERLAP = json.loads((paths.directory / "bmr.countries_overlap.json").read_text())
+COUNTRIES_OVERLAP = [{int(k): set(v) for k, v in overlaps.items()} for overlaps in COUNTRIES_OVERLAP]
+
+# REGION AGGREGATES
+REGIONS = {
+    "Africa": {},
+    "Asia": {},
+    "North America": {},
+    "South America": {},
+    "Europe": {},
+    "Oceania": {},
+}
+
+
+def run(dest_dir: str) -> None:
+    #
+    # Load inputs.
+    #
+    # Load meadow dataset.
+    ds_meadow = paths.load_dataset("bmr")
+    ds_regions = paths.load_dataset("regions")
+    ds_population = paths.load_dataset("population")
+
+    # Read table from meadow dataset.
+    tb = ds_meadow["bmr"].reset_index()
+
+    #
+    # Process data.
+    #
+    # Harmonize country names
+    tb = geo.harmonize_countries(df=tb, countries_file=paths.country_mapping_path)
+
+    # Identify observations from countries that would overlap with other countries (Russia with USSR in 1991)
+    tb = remove_country_overlaps(tb)
```

Comment:
Maybe a sense check here to check you've removed the amount of rows that you expect to?